src/token_retriever/credential.rs

```diff
@@ -13,41 +13,53 @@ pub(super) const DEFAULT_JWT_CLAIM_EXP: TimeDelta = TimeDelta::seconds(180);
 /// The "aud" (audience) claim identifies the recipients that the JWT is intended for.
 pub(super) const DEFAULT_AUDIENCE: &str = "https://www.newrelic.com/";
 
+pub trait AuthCredentialBuilder {
+    fn build_request_auth_credential(
```

Comment:
I was expecting to see this used somewhere, maybe on a subsequent PR?

Reply:
It is used [here](https://github.com/newrelic/newrelic-auth-rs/pull/99/files#diff-0e54f7cad00e3e5b5a631539a01e04e2979a9d75de3942ad4a25fe9a659868afR21). The change is a bit subtle because I kept the same name for the method